backend/src/dao.py

```diff
@@ -287,4 +290,36 @@ def week_species_summary(db: connection) -> dict[str, list]:
                 }
             except psycopg2.Error as e:
                 print("Error executing SQL query:", e)
-                raise HTTPException(status_code=500, detail="Database error")
\ No newline at end of file
+                raise HTTPException(status_code=500, detail="Database error")
+
+
+    
+    @staticmethod
+    def node_health_check(db: connection) -> list:
+        """Returns the time of the last message from each node along with the type of node"""
+        @dataclass
+
+        class NodeReport:
+            """Node health report query object"""
+            latest_time: datetime
+            ndescription: str
+            ntype: str
+        with db.cursor() as curs:
+            try:
+                curs.execute(sql.SQL(
+                    """
+                    select latest_time, n.ndescription, n.ntype from  (
+                        select max(t.ttime) as latest_time, n.nid from timestampindex t natural inner join node n 
+                        group by  n.nid
+                    )
+                    natural inner join node n
+                    order by n.ntype 
```

Comment:
Keep all SQL syntax all caps for improved legibility. 
```suggestion
                    SELECT latest_time, n.ndescription, n.ntype FROM  (
                        select max(t.ttime) AS latest_time, n.nid FROM timestampindex t NATURAL INNER JOIN node n 
                        GROUP BY n.nid
                    )
                    NATURAL INNER JOIN node n
                    ORDER BY n.ntype 
```

Reply:
also we can simplify this query by removing the second join. 


```suggestion from deepsek lmao
SELECT t.max_time AS latest_time, n.ndescription, n.ntype
FROM node n
JOIN (
    SELECT nid, MAX(ttime) AS max_time
    FROM timestampindex
    GROUP BY nid
) t ON n.nid = t.nid
ORDER BY n.ntype;
```